tests/rscript_parse_time.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+# Added 2017-10-28 by Stephen Workman, released under ASL 2.0
+
+getts="python $srcdir/rscript_parse_time_get-ts.py"
+
+$getts selftest
+
+if [[ $? -ne 0 ]]; then
+  printf "Failed own self-test(s)!\n"
+  . $srcdir/diag.sh error-exit 1
+fi
+
+rfc3164_1="Oct  5 01:10:11"
+rfc3164_1_r=$($getts "$rfc3164_1")
+
+rfc3164_2="Jan 31 13:00:00"
+rfc3164_2_r=$($getts "$rfc3164_2")
+
+rfc3164_3="Feb 28 14:35:00"
+rfc3164_3_r=$($getts "$rfc3164_3")
+
+rfc3164_4="Mar  1 14:00:00"
+rfc3164_4_r=$($getts "$rfc3164_4")
+
+rfc3164_5="Apr  3 15:00:00"
+rfc3164_5_r=$($getts "$rfc3164_5")
+
+rfc3164_6="May  5 16:00:00"
+rfc3164_6_r=$($getts "$rfc3164_6")
+
+rfc3164_7="Jun 11 03:00:00"
+rfc3164_7_r=$($getts "$rfc3164_7")
+
+rfc3164_8="Jul 15 05:00:00"
+rfc3164_8_r=$($getts "$rfc3164_8")
+
+rfc3164_9="Aug 17 08:00:00"
+rfc3164_9_r=$($getts "$rfc3164_9")
+
+rfc3164_10="Sep 20 18:00:00"
+rfc3164_10_r=$($getts "$rfc3164_10")
+
+rfc3164_11="Nov 23 19:00:00"
+rfc3164_11_r=$($getts "$rfc3164_11")
+
+rfc3164_12="Dec 25 20:00:00"
+rfc3164_12_r=$($getts "$rfc3164_12")
+
+. $srcdir/diag.sh init
+. $srcdir/diag.sh generate-conf
+. $srcdir/diag.sh add-conf '
+module(load="../plugins/imtcp/.libs/imtcp")
+module(load="../plugins/omstdout/.libs/omstdout")
+input(type="imtcp" port="13514")
+
+# $DebugLevel 2
+
+# RFC 3164 Parse Tests
+set $!datetime!rfc3164_1  = parse_time("'"$rfc3164_1"'");
+set $!datetime!rfc3164_2  = parse_time("'"$rfc3164_2"'");
+set $!datetime!rfc3164_3  = parse_time("'"$rfc3164_3"'");
+set $!datetime!rfc3164_4  = parse_time("'"$rfc3164_4"'");
+set $!datetime!rfc3164_5  = parse_time("'"$rfc3164_5"'");
+set $!datetime!rfc3164_6  = parse_time("'"$rfc3164_6"'");
+set $!datetime!rfc3164_7  = parse_time("'"$rfc3164_7"'");
+set $!datetime!rfc3164_8  = parse_time("'"$rfc3164_8"'");
+set $!datetime!rfc3164_9  = parse_time("'"$rfc3164_9"'");
+set $!datetime!rfc3164_10 = parse_time("'"$rfc3164_10"'");
+set $!datetime!rfc3164_11 = parse_time("'"$rfc3164_11"'");
+set $!datetime!rfc3164_12 = parse_time("'"$rfc3164_12"'");
+
+# RFC 3339 Parse Tests
+set $!datetime!rfc3339    = parse_time("2017-10-05T01:10:11Z");
+set $!datetime!rfc3339tz1 = parse_time("2017-10-05T01:10:11+04:00");
+set $!datetime!rfc3339tz2 = parse_time("2017-10-05T01:10:11+00:00");
+
+set $!datetime!inval1 = parse_time("not a date/time");
+set $!datetime!inval2 = parse_time("2017-10-05T01:10:11");
+set $!datetime!inval3 = parse_time("2017-SOMETHING: 42");
+
+template(name="outfmt" type="string" string="%!datetime%\n")
+local4.* action(type="omfile" file="rsyslog.out.log" template="outfmt")
+local4.* :omstdout:;outfmt
+'
+
+. $srcdir/diag.sh startup
+. $srcdir/diag.sh tcpflood -m1 -y | sed 's|\r||'
```

Comment:
what is the intent here? (in line 87)

Reply:
You mean the `sed 's|\r||'`? `tcpflood` produces a carriage return in its output (see below), it was bugging me :)

```
^M00000001 messages sent
```

the `sed` changes it to

```
00000001 messages sent
```

Reply:
I would suggest to remove that as it may make other stumble over it as well. The problem in travis is that it mimics a tty, so that tcpflood's supression logic for non-ttys does not kick in.

Reply:
You're da boss ;)

Done.